apps/server/src/modules/account/uc/account.uc.spec.ts

```diff
@@ -913,6 +936,82 @@ describe('AccountUc', () => {
 				accountUc.findAccountById({ userId: mockSuperheroUser.id } as ICurrentUser, { id: 'xxx' } as AccountByIdParams)
 			).rejects.toThrow(EntityNotFoundError);
 		});
+		it('should throw, if target account has no user', async () => {
+			await expect(
+				accountUc.findAccountById({ userId: mockSuperheroUser.id } as ICurrentUser, { id: 'xxx' } as AccountByIdParams)
+			).rejects.toThrow(EntityNotFoundError);
+		});
+	});
+
+	describe('createAccount', () => {
+		afterEach(() => {
+			jest.clearAllMocks();
+		});
+
+		it('should sanitize username for local user', async () => {
+			const spy = jest.spyOn(accountService, 'create');
+			const params: AccountCreateParams = {
+				username: ' John.Doe@domain.tld ',
+			};
+			await accountUc.createAccount(params);
+			expect(spy).toHaveBeenCalledWith({
+				username: 'john.doe@domain.tld',
+			});
+		});
+		it('should not sanitize username for external user', async () => {
+			const spy = jest.spyOn(accountService, 'create');
+			const params: AccountCreateParams = {
+				username: ' John.Doe@domain.tld ',
+				systemId: 'ABC123',
+			};
+			await accountUc.createAccount(params);
+			expect(spy).toHaveBeenCalledWith(
+				expect.objectContaining({
+					username: ' John.Doe@domain.tld ',
+				})
+			);
+		});
+		it('should throw if username for a local user is not an email', async () => {
+			const params: AccountCreateParams = {
+				username: 'John Doe',
+			};
+			await expect(accountUc.createAccount(params)).rejects.toThrow('Username is not an email');
+		});
+		it('should not throw if username for an external user is not an email', async () => {
+			const params: AccountCreateParams = {
+				username: 'John Doe',
+				systemId: 'ABC123',
+			};
+			await expect(accountUc.createAccount(params)).resolves.not.toThrow();
+		});
+		it('should throw if account already exists', async () => {
+			const params: AccountCreateParams = {
+				username: mockStudentUser.email,
+				userId: mockStudentUser.id,
+			};
+			await expect(accountUc.createAccount(params)).rejects.toThrow('Account already exists');
+		});
+		it('should throw if username already exists', async () => {
```

Comment:
Not necessary for coverage, but checking that it is only searching in the same system would be nice to have.

Reply:
We can treat `undefined` as another system, so we have implicitly tested for uniqueness in the same system.